### PR TITLE
[SPARK-48814][BUILD] Upgrade `tink` to 1.14.0

### PR DESCRIPTION
**Pull request by wayneguow**
### What changes were proposed in this pull request?

This PR aims to upgrade `tink` from 1.13.0 to 1.14.0.

### Why are the changes needed?

There are some bug fixes and performance improvments:
https://github.com/tink-crypto/tink-java/releases/tag/v1.14.0

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass GA.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2774,6 +2774,10 @@
             <groupId>com.google.http-client</groupId>
             <artifactId>google-http-client</artifactId>
           </exclusion>
+          <exclusion>
```
Comment (LuciferYang) [inline pom.xml:2777]: Why does upgrading `tink` require adding this `exclusion`?

Comment (wayneguow) [reply pom.xml:2777]: Cause that `tink` upgrade `protobuf-java` to 4.27.0 in version 1.14.0, but it's not compatible with 3.25.1. (related protobuf issues: https://github.com/protocolbuffers/protobuf/issues/16452, https://github.com/protocolbuffers/protobuf/issues/17247)

And I notice that there are some users in `tink` community also facing this problem, let's wait for the community to provide a solution(https://github.com/tink-crypto/tink-java/issues/31 and https://github.com/tink-crypto/tink-java/pull/38). @LuciferYang

Comment (LuciferYang) [reply pom.xml:2777]: If that's the case, I suggest skipping this version.

Comment (wayneguow) [reply pom.xml:2777]: Thanks for the suggestion. I will close this PR first and keep watching on this protobuf realted issue in the future.

**Pull request closed without merging**